concordium-consensus/src/Concordium/GlobalState/PurgeTransactions.hs

```diff
@@ -175,8 +190,11 @@ purgeTables lastFinCommitPoint oldestArrivalTime currentTime TransactionTable{..
             !ptt1 = ptt0 & pttUpdates . at' uty %~ updptt mmax
         in  (nfcu{_nfcuMap = newNFCUMap}, (ptt1, uis1))
     purge = do
-        -- Purge each account
-        nnft <- HM.traverseWithKey purgeAccount _ttNonFinalizedTransactions
+        -- Purge each account, and possibly remove the @AccountNonFinalizedTransactions@
+        -- if an account does not have any pending transactions left.
+        s <- get
+        let (nnft, s') = HM.foldlWithKey' purgeAccount (HM.empty, s) _ttNonFinalizedTransactions
+        put s'
```

Comment:
**Minor thing:** Please introduce a variable name instead of just `s` and `s'`.

I don't think it is obvious what this is, and it would save the reader some time instead of having to build up enough context.

Reply:
I don't agree, but I added some more words. Hope it's clear now. It's a "complicated"
 state in the sense that it's a tuple where the members each have long names. 
 
But I do think it's already clear when looking at the type signature for `purgeAccount` (which you probably look at anyway) - also the lsp server also tells it.